Gem/Code/Source/NetSoakTestSystemComponent.cpp

```diff
@@ -176,7 +180,18 @@ namespace NetSoakTest
 
     void NetSoakTestSystemComponent::OnTick(float deltaTime, [[maybe_unused]] AZ::ScriptTimePoint time)
     {
-        [[maybe_unused]] auto elapsedMs = aznumeric_cast<AZ::TimeMs>(aznumeric_cast<int64_t>(deltaTime / 1000.0f));
+    	AZ::TimeMs elapsedMs = aznumeric_cast<AZ::TimeMs>(aznumeric_cast<int64_t>(deltaTime / 1000.0f));
+
+        m_totalElapsedTime += elapsedMs;
```

Comment:
Instead of accumulating this value every frame along with any error it may contain, I recommend setting a future time-point for when the timeout is triggered. The boolean check below would then become something like `(time > m_soakEndTimepoint)`

Reply:
Inverted check